### PR TITLE
strtod test

### PR DESCRIPTION
**Pull request by DawidSzpejna**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work: https://github.com/phoenix-rtos/libphoenix/pull/193
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by nalajcie** [comment]
Did not review, I've just run Your tests on host (with glibc) and one of the tests fails - please verify if it's correct:
image

Also - please change order of params to `TEST_ASSERT_*` macros:
image

So the error messages would make more sense :)

File: libc/strtod.c [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+TEST(strtod_family, errno_test)
+{
+	errno = 0;
+	const char num1[] =
+		"17976931348623157081452742373170435679807056752584499659891747680315"
+		"72607800285387605895586327668781715404589535143824642343213268894641"
+		"82768467546703537516986049910576551282076245490090389328944075868508"
+		"45513394230458323690322294816580855933212334827479782620414472316873"
+		"8177180919299881250404026184124858368123111";
+	strtod(num1, NULL);
```
Comment (agkaminski) [inline libc/strtod.c:154]: Hmm, shouldn't we also assert that `strtod` retval is `NAN`?

**Comment by nalajcie** [comment]
Please update with upstream now that we have new `strtod` implementation which should pass the tests

File: libc/strtod.c (hunk 1 above)
Comment (DawidSzpejna) [reply libc/strtod.c:154]: It is very reasonable to check retval, i will add it. However, i think retval should be equal to `\0` character.

File: libc/strtod.c [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+		"27976931348623157081452742373170435679807056752584499659891747680315"
+		"72607800285387605895586327668781715404589535143824642343213268894641"
+		"82768467546703537516986049910576551282076245490090389328944075868508"
+		"45513394230458323690322294816580855933212334827479782620414472316873"
+		"8177180919299881250404026184124858368";
+	TEST_ASSERT_DOUBLE_IS_INF(strtod(num3, &retval));
+	TEST_ASSERT_EQUAL_STRING("", retval);
+}
+
+TEST(strtod_family, errno_test)
```
Comment (niewim19) [inline libc/strtod.c:154]: Why splitting into `errno_test` and `too_long_numbers`? The lateral should also cause errno to be set, we can check it there without separate test group. There is no scenario where conversion fails and errno is not set.

Comment (DawidSzpejna) [reply libc/strtod.c:154]: I decided to split into `errno_test` and `too_long_numbers` because `strtod` may fail with `EINVAL` if no conversion could be performed.

**Review by niewim19 — COMMENTED** [review]
Not all edge cases are covered, but this would be a good task for our test-team @damianloew  

For me it is good to go, please update this branch to current version. I am adding @damianloew as reviewer.

**Comment by damianloew** [comment]
> Not all edge cases are covered, but this would be a good task for our test-team @damianloew
> 
> For me it is good to go, please update this branch to current version. I am adding @damianloew as reviewer.

Yeah, we will upgrade this test to provide a better coverage, but let's merge the current version for now (of course after resolving conflicts).
